Modules/DataModel/Sources/PocketCastsDataModel/Public/DataManager.swift

```diff
@@ -90,11 +90,26 @@ public class DataManager {
         let userEpisodes = userEpisodeManager.allUpNextEpisodes(dbQueue: dbQueue)
 
         // this extra step is to make sure we return the episodes in the order they are in the up next list, which they won't be if there's both Episodes and UserEpisodes in Up Next
+        if userEpisodes.isEmpty {
+            return episodes
+        }
         var convertedEpisodes = [BaseEpisode]()
+        var episodeIndex = 0
+        var userEpisodeIndex = 0
         for upNextEpisode in allUpNextEpisodes {
-            guard let episode: BaseEpisode = episodes.first(where: { $0.uuid == upNextEpisode.episodeUuid }) ?? userEpisodes.first(where: { $0.uuid == upNextEpisode.episodeUuid }) else { continue }
-
-            convertedEpisodes.append(episode)
+            if let episode = episodes[safe: episodeIndex] {
+                if episode.uuid == upNextEpisode.episodeUuid {
+                    convertedEpisodes.append(episode)
+                    episodeIndex += 1
+                    continue
+                }
+            }
```

Comment:
You can have the two conditions in a single `if`:

```suggestion
            if let episode = episodes[safe: episodeIndex],
               episode.uuid == upNextEpisode.episodeUuid {
                convertedEpisodes.append(episode)
                episodeIndex += 1
                continue
            }
```

Reply:
WOW! Swift's syntax can still surprise me haha, thanks for this unexpected lesson!